django/cantusdb_project/main_app/views/source.py

```diff
@@ -327,27 +337,35 @@ def get_queryset(self) -> QuerySet[Source]:
             # provenance_q = Q()
             summary_q = Q()
 
-            # For each term, add it to the Q object of each field with an OR operation.
-            # We split the terms so that the words can be separated in the actual
-            # field, allowing for a more flexible search, and a field needs
-            # to match only one of the terms
-            for term in general_search_terms:
-                holding_institution_q |= Q(holding_institution__name__icontains=term)
+            # Add unquoted terms to the Q object with partial matching (icontains)
+            for term in unquoted_terms:
+                holding_institution_q |= Q(
+                    holding_institution__name__unaccent__icontains=term
+                )
                 holding_institution_city_q |= Q(
-                    holding_institution__city__icontains=term
+                    holding_institution__city__unaccent__icontains=term
                 )
                 shelfmark_q |= Q(shelfmark__unaccent__icontains=term)
                 siglum_q |= Q(holding_institution__siglum__unaccent__icontains=term)
                 description_q |= Q(description__unaccent__icontains=term)
                 summary_q |= Q(summary__unaccent__icontains=term)
-                name_q |= Q(name__icontains=term)
-                # provenance_q |= Q(provenance__name__icontains=term)
-            # All the Q objects are put together with OR.
-            # The end result is that at least one term has to match in at least one
-            # field
-            # general_search_q = (
-            #     title_q | siglum_q | description_q | provenance_q
-            # )
+                name_q |= Q(name__unaccent__icontains=term)
+
+            # Add quoted terms to the Q object with exact matching (iexact)
+            for term in quoted_terms:
+                holding_institution_q |= Q(
+                    holding_institution__name__unaccent__iexact=term
```

Comment:
I don't think `iexact` is correct here...

If someone searches for "North Vancouver", don't I want any results where "North Vancouver" shows up somewhere in one of these fields, not just results where a field is exactly "North Vancouver"? 

I think `contains` is probably still the method, it's just we're not searching single words, but sometimes groups of words depending on the quotes.